personalize.api.php

```diff
@@ -56,15 +56,14 @@ function hook_personalize_option_set_type() {
 }
 
 /**
- * An agent is about to be created or updated.
- *
- * This hook is invoked before the agent is saved to the database so
- * modules can alter the information that is saved.
+ * Allows the alteration of an agent before it is saved.
  *
  * @param stdClass $agent
  *   A stdClass object representing the agent being saved.
+ * @param array $prior_values
+ *   The unalterable prior data for the agent before this save.
  */
-function hook_personalize_agent_presave($agent) {
+function hook_personalize_agent_alter(&$agent, $prior_values) {
```

Comment:
Well - first presave actually wasn't being called in the personalize module anyway.  But when I looked at adding it in I remembered that module_invoke passes by value and not by reference.